src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/Engine/NodeOperations.cs

```diff
@@ -2,21 +2,44 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System.Collections.Generic;
+using System;
+using Microsoft.CodeAnalysis.Collections;
 using Microsoft.CodeAnalysis.Formatting.Rules;
+using Microsoft.CodeAnalysis.PooledObjects;
 
 namespace Microsoft.CodeAnalysis.Formatting
 {
     /// <summary>
     /// this collector gathers formatting operations that are based on a node
     /// </summary>
-    internal class NodeOperations
+    internal record NodeOperations : IDisposable
```

Comment:
1. consider sealing.
2. does being a record actually help here at all?

Reply:
Bah, the record change was from when I was trying something else out. Reverting.